File: PI24-mainnnnn/src/main/java/tn/esprit/piproject/Entities/Request.java
package tn.esprit.piproject.Entities;

import jakarta.persistence.Id;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "requests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
public class Request {
    @Id
    private int id ;
    private String message;
    @DBRef
    private Offer offer ;
    private  boolean submit ;
    @DBRef
    private  User Student ;
    @DBRef
    private  User supervisor ;
}

File: PI24-mainnnnn/AngularPIFrontnnnnn/src/app/core/Request.ts
import { Company } from 'src/app/core/Company';
import { Offer } from "./Offer";
import { User } from "./User";

export class Request {
    id? : Number;
    submit : boolean;
    message : String;
    offer : Offer;
    student : User;
    supervisor : User;
}
